src/util/dynamicloggingcategory.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <QByteArray>
+#include <QLoggingCategory>
+
+/// DynamicLoggingCategory is a wrapper around QLoggingCategory
+/// that facilitates using a QLoggingCategory for a category
+/// name string that is determined at run time. If the category
+/// name is fully known at compile time, simply construct a
+/// QLoggingCategory with a string literal.
+///
+/// DynamicLoggingCategory stores the logging category name as a
+/// QByteArray. This is helpful because QLoggingCategory has a deleted
+/// copy constructor, does not take ownership nor make a copy of the
+/// const char* passed to its constructor, and requires that the
+/// data pointed to by the const char* lives as long as the
+/// QLoggingCategory.
+///
+/// DynamicLoggingCategory takes advantage of QByteArray's
+/// implicit sharing to ensure that the const char* for the
+/// QLoggingCategory lives as long as the QLoggingCategory without
+/// requiring excessive deep copies.
```

Comment:
The documentation is longer than the implementation :grimacing: I wish QLoggingCategory just took care of this for us.

Reply:
They probably adopted QLoggingCategory straight from their embedded or automotive divisions where everything is static. Its design doesn't really fit for more general use cases.

Reply:
I'll report a bug on the Qt bug tracker about this clumsy API after we merge this.